src/lib.rs

```diff
@@ -128,7 +141,7 @@ pub fn pack_rects<
     box_size_heuristic: &BoxSizeHeuristicFn,
     more_suitable_containers_fn: &ComparePotentialContainersFn,
 ) -> Result<RectanglePackOk<RectToPlaceId, BinId>, RectanglePackError> {
-    let mut packed_locations = HashMap::new();
+    let mut packed_locations = BTreeMap::new();
```

Comment:
Same thing in this file, `KeyValMap` and make it use `HashMap` in std and `BTreeMap` in `no_std`.

You can just define `KeyValMap` once in the crate root and everywhere else that's applicable use `crate::KeyValMap`